### ISSUE TITLE
contain-facebook has problems dealing with forks, like contain-goolge/amazon/microsoft

### ISSUE THREAD

**Issue by evertonstz** [mozilla/contain-facebook#636]
The problem is pretty annoying and arrives once using contain-facebook with forks, like contain-goolge/amazon/microsoft. Problems arrives in situations like this one:

Assume this person is using contain-facebook and forks.
Let's say someone is browsing Facebook (it's contained, ofc) and finds a google link in there, if this person tries to open this google link in a new tap with middle mouse click or using the "open link in a new Facebook tab", this will result in opening two contain-google tabs instead of just one. This happens using any of the doesn't matter if contain-facebook is the opening or the one being opened.

**Comment by maxxcrawford** [comment]
With any of the Contain-Google/Amazon/Microsoft extensions, those extensions dictate which container a specific link should be opened in. I am confused on why you'd want to open a Google link in the Facebook container, when you have both Contain-X extensions installed. 

I'd recommend using [Multi-Account Containers](https://addons.mozilla.org/en-US/firefox/addon/multi-account-containers/) to manage multiple containers at once. Mozilla doesn't manage any other Contain-X extension besides this one, but we do support cross-compatibility for Multi-Account Containers and Facebook Container.

**Comment by evertonstz** [comment]
> With any of the Contain-Google/Amazon/Microsoft extensions, those extensions dictate which container a specific link should be opened in. I am confused on why you'd want to open a Google link in the Facebook container, when you have both Contain-X extensions installed.

Because that's not what I'm reporting as a bug and I'm not looking to open google links in facebook containers. The extensions are having the intended behavior: opening a google link from a facebook page will open the google in a new tab with the correct google container, the bug I'm reporting is that for some reason it opens two tabs instead of a single one, as it should be expected.

Since an image speaks more than a thousand words, [this](https://imgur.com/a/wusAd9U) is what is happening, middle-click once on the facebook result, got two facebook tabs instead of a single one.

Hope I made things clearer now.

**Comment by maxxcrawford** [comment]
@evertonstz Haha. Truely! Thanks for the image. 

TLDR; Middle-click a FB link from a non-FB container opens up two FB tabs.

**Comment by evertonstz** [comment]
> @evertonstz Haha. Truely! Thanks for the image.
> 
> TLDR; Middle-click a FB link from a non-FB container opens up two FB tabs.

No problems! English is not my first language so I sometimes have problems explaining this kind of things. I'll add to your TLDR that's it happens between on all extensions: FB > Google, Google > FB, Google > Microsoft, Microsoft > Amazon. It won't happen to sites that don't have a "contain" extension ofc, so opening, let's say, Github from Google everything works fine.

Cheers!

**Comment by shbach** [comment]
> > @evertonstz Haha. Truely! Thanks for the image.
> > TLDR; Middle-click a FB link from a non-FB container opens up two FB tabs.
> 
> No problems! English is not my first language so I sometimes have problems explaining this kind of things. I'll add to your TLDR that's it happens between on all extensions: FB > Google, Google > FB, Google > Microsoft, Microsoft > Amazon. It won't happen to sites that don't have a "contain" extension ofc, so opening, let's say, Github from Google everything works fine.
> 
> Cheers!

To add to this, it isn't just middle click. In general, opening FB from a non-FB container (whether by middle click, left click, typing in the url bar, clicking a FB bookmark, etc) will result in two FB tabs.

**Comment by maxxcrawford** [comment]
@shbach Is this in [multi-account containers](https://addons.mozilla.org/en-US/firefox/addon/multi-account-containers/) or a different container extension?

**Comment by shbach** [comment]
> @shbach Is this in [multi-account containers](https://addons.mozilla.org/en-US/firefox/addon/multi-account-containers/) or a different container extension?

@maxxcrawford Sorry, I could have been more clear. When I said container extension I specifically meant the FB container extension or any extensions that are a fork of it. This has nothing to do with the multi-account containers extension. 

I've only personally tested this issue with the [Google Container](https://addons.mozilla.org/en-US/firefox/addon/google-container/) extension but it should apply to all the forks according to @evertonstz. To see this issue yourself, have the google container extension installed together with the FB container extension. From a FB tab try to navigate to a google page, or from a google tab navigate to a FB page and it will result in two identical tabs. Hopefully that made sense. Let me know if you need additional clarification.

**Comment by scapeh1ps** [comment]
I'm bumping this thread because it's still not fixed. For eg. I cannot use google container with Facebook container, opening a Facebook/Instagram url from google contained tab opens double fb tabs.

**Comment by bolokanar** [comment]
This is a pretty annoying bug and needs to be finally looked into.
